application/libraries/Grocery_CRUD.php

```diff
@@ -2574,25 +2656,150 @@ protected function get_relation_n_n_input($field_info_type, $selected_values)
+
+				if(!empty($unselected_values))
+				foreach($unselected_values as $id => $name){
+					$input .= "<option value='$id'>$name</option>";
+				}
+
+				if(!empty($selected_values))
+				foreach($selected_values as $id => $name){
+					$input .= "<option value='$id'>$name</option>";
+				}
+
+				$input .= "</select>";
+
+				$extra_fields = $this->basic_model->get_field_types($field_info_type->extras->relation_table);
+				$extra_state_info = $this->getStateInfo();
+
+				$input .= "<div id='{$field_info_type->name}Tabs' class='tabs'>";
+				$input .= " <ul>";
+				if(!empty($selected_values))
+				foreach($selected_values as $id => $name){
+					$input .= "<li><a href='#{$field_info_type->name}-relation-$id'>$name</a><span class='ui-icon ui-icon-close' data-value='$id' data-field='{$field_info_type->name}' role='presentation'>Remove Tab</span></li>";
+				}
+
+				$input .= " </ul>";
+				if(!empty($selected_values))
+				foreach($selected_values as $id => $name){
+					$extra_where = array($field_info_type->extras->primary_key_alias_to_this_table => $extra_state_info->primary_key ,
+							$field_info_type->extras->primary_key_alias_to_selection_table => $id);
+					$values = $this->basic_model->get_extra_edit_values($field_info_type->extras->relation_table, $extra_where);
+
+					$input .= "<div id='{$field_info_type->name}-relation-$id'><p></p>";
+
+					foreach ($extra_fields as $extra_field){
+						if ( $this->unset_edit_fields !== null
+						&& is_array($this->unset_edit_fields)
+						&& in_array(('extra_field_'.$extra_field->name),$this->unset_edit_fields)
+						) {
+							continue;
+						}
+						if( $extra_field->name != $field_info_type->extras->primary_key_alias_to_this_table
+						&& $extra_field->name != $field_info_type->extras->primary_key_alias_to_selection_table){
+
+							$extra_field->db_type = $extra_field->type;
+							$extra_field->db_max_length = $extra_field->max_length;
+
+							$extra_field_info = (object)array();
+							$extra_field_info->name		= $field_info_type->name.'Extras['.$id.']['.$extra_field->name.']';
+							$extra_field_info->crud_type 	= $this->get_type($extra_field);
+							$extra_field_info->type = $extra_field->type;
+							if($extra_field_info->crud_type== 'text'){
+								$extra_field_info->extras = 'text_editor';
+							}
+							elseif($extra_field_info->crud_type == 'set' || $extra_field_info->crud_type == 'enum'){
+								$extra_field_info->extras = $this->basic_model->enum_select($field_info_type->extras->relation_table, $extra_field->name);
+							}
+							else{
+								$extra_field_info->extras = null;
+							}
+							$extra_field_info->db_max_length = $extra_field->db_max_length;
+							$extra_field_info->required	= false;
+							$extra_field_info->display_as = ucfirst(str_replace("_"," ",$extra_field->name));
+							$input .= "<div>".$extra_field_info->display_as."</div><div>".$this->get_field_input($extra_field_info, $values->{$extra_field->name})->input."</div>";
```

Comment:
//Add callback functionality 
							if(!isset($this->callback_edit_field[$extra_field->name])){
								$input .= "<div>".$extra_field_info->display_as."</div><div>".$this->get_field_input($extra_field_info, $values->{$extra_field->name})->input."</div>";
							} else {
								$input .= call_user_func($this->callback_edit_field[$extra_field->name], $values->{$extra_field->name}, $id, $extra_field_info);
							}